File: src/longest-unequal-adjacent-groups-subsequence-i/longest-unequal-adjacent-groups-subsequence-i.py
class Solution(object):
    def getLongestSubsequence(self, words, groups):
        """
        :type words: List[str]
        :type groups: List[int]
        :rtype: List[str]
        """
        result = [words[0]]
        for i in range(1, len(words)):
            if groups[i] != groups[i-1]:
                result += [words[i]]
        return result

File: src/longest-unequal-adjacent-groups-subsequence-i/longest-unequal-adjacent-groups-subsequence-i.java
class Solution {
    public List<String> getLongestSubsequence(String[] words, int[] groups) {
        List<String> result = new ArrayList<>();
        result.add(words[0]);
        for (int i = 1; i < words.length; i++) {
            if (groups[i] != groups[i-1]) {
                result.add(words[i]);
            }
        }
        return result;
    }
}